server/src/middlewares/middleware.js

```diff
@@ -93,6 +96,56 @@ class middleware {
     }
     return next();
   }
+
+  static async validateUser(req, res, next) {
+    const {
+      firstName, lastName, email, password,
+    } = req.body;
+    if (!firstName && !lastName && !email && !password) {
+      return res.status(400).send({
+        error: 'Sorry, please all field is required!',
+      });
+    }
+    const schema = Joi.object().keys({
+      firstName: Joi.string().strict().trim().min(2)
+        .required()
+        .error(() => 'firstname is required'),
+      lastName: Joi.string().strict().trim().min(2)
+        .required()
+        .error(() => 'lastname is required'),
+      email: Joi.string().strict().trim().min(8)
```

Comment:
Using a minimum of 8 characters for the email input? What if the user's name isn't long enough to have up to 8 characters in the email?
I think the minimum characters should be reduced and instead, a regex used to ensure it's in an email format

Reply:
👍